server/data/educationAndWorkPlanApiMapper.test.ts

```diff
@@ -0,0 +1,43 @@
+import type { AddStepDto, CreateGoalDto } from 'dto'
+import type { CreateGoalRequest, CreateStepRequest } from 'educationAndWorkPlanApiClient'
+import moment from 'moment'
+import EducationAndWorkPlanApiMapper from './educationAndWorkPlanApiMapper'
+
+describe('educationAndWorkPlanApiMapper', () => {
+  it.skip('should map to CreateGoalDto given valid form data', () => {
+    // Given
+    const educationAndWorkPlanApiMapper = new EducationAndWorkPlanApiMapper()
+    const prisonNumber = 'A1234BC'
+    const addStepDto = {
+      title: 'Book Spanish course',
+      targetDate: moment('2123-01-31', 'YYYY-MM-DD').toDate(),
+    } as AddStepDto
+    const createGoalDto = {
+      prisonNumber,
+      title: 'Learn Spanish',
+      reviewDate: moment('2123-06-31', 'YYYY-MM-DD').toDate(),
+      steps: [addStepDto],
+      note: 'Prisoner is not good at listening',
+    } as CreateGoalDto
+
+    const expectedAddStepRequest = {
+      title: 'Book Spanish course',
+      targetDate: moment('2123-01-31', 'YYYY-MM-DD').toDate(),
+      sequenceNumber: 1,
+    } as CreateStepRequest
+    const expectedCreateGoalRequest = {
+      prisonNumber,
+      title: 'Learn Spanish',
+      category: 'WORK',
+      reviewDate: moment('2123-06-31', 'YYYY-MM-DD').toDate(),
+      steps: [expectedAddStepRequest],
+      note: createGoalDto.note,
+    } as CreateGoalRequest
+
+    // When
+    const createGoalRequest = educationAndWorkPlanApiMapper.createGoalDtoToCreateGoalRequest(createGoalDto)
+
+    // Then
+    expect(createGoalRequest).toEqual(expectedCreateGoalRequest)
```

Comment:
I cant be sure, but it might be because the objects you are comparing are "deep" / ie. not simple flat objects
I think theres a `toDeepEquals` or similarly named method you might be able to try?

Reply:
Unfortunately that doesn't exist and I've tried a few different methods (like `toStrictEqual`), but no joy so far...

Reply:
I'm using `JSON.stringify` for now and have enabled the test.

Reply:
I've found the problem; and you are absolutely gonna kick yourself!!!! (though I blame this on one of Javascript's weirder parts; of which there are many!)
I'll talk you through it in a huddle this morning 👍